engine/src/main/java/org/pentaho/di/trans/steps/edi2xml/grammar/FastSimpleGenericEdifactDirectXMLParser.java

```diff
@@ -0,0 +1,1096 @@
+
+  public static final int EOF = -1;
+  public static final int T__9 = 9;
+  public static final int T__10 = 10;
+  public static final int T__11 = 11;
+  public static final int T__12 = 12;
+  public static final int T__13 = 13;
+  public static final int T__14 = 14;
+  public static final int COMPLEX_ELEMENT_ITEM_SEPARATOR = 4;
+  public static final int ELEMENT_SEPARATOR = 5;
+  public static final int RELEASE_CHARACTER = 6;
+  public static final int SEGMENT_TERMINATOR = 7;
+  public static final int TEXT_DATA = 8;
+
+  // delegates
+  public Parser[] getDelegates() {
+    return new Parser[] {};
+  }
+
+  // delegators
+
+  public FastSimpleGenericEdifactDirectXMLParser( TokenStream input ) {
+    this( input, new RecognizerSharedState() );
+  }
+
+  public FastSimpleGenericEdifactDirectXMLParser( TokenStream input, RecognizerSharedState state ) {
+    super( input, state );
+  }
+
+  protected StringTemplateGroup templateLib = new StringTemplateGroup(
+    "FastSimpleGenericEdifactDirectXMLParserTemplates", AngleBracketTemplateLexer.class );
+
+  public void setTemplateLib( StringTemplateGroup templateLib ) {
+    this.templateLib = templateLib;
+  }
+
+  public StringTemplateGroup getTemplateLib() {
+    return templateLib;
+  }
+
+  /**
+   * allows convenient multi-value initialization: "new STAttrMap().put(...).put(...)"
+   */
+  public static class STAttrMap extends HashMap<String, Object> {
+    public STAttrMap put( String attrName, Object value ) {
+      super.put( attrName, value );
+      return this;
+    }
+
+    public STAttrMap put( String attrName, int value ) {
+      super.put( attrName, new Integer( value ) );
+      return this;
+    }
+  }
+
+  public String[] getTokenNames() {
+    return FastSimpleGenericEdifactDirectXMLParser.tokenNames;
+  }
+
+  public String getGrammarFileName() {
+    return "C:\\workspace-sts\\Kettle trunk - "
+      + "restruct\\engine\\src\\org\\pentaho\\di\\trans\\steps\\edi2xml\\grammar\\"
+      + "FastSimpleGenericEdifactDirectXML.g";
+  }
+
+  public static final String XML_HEAD = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n";
+  public static final String TAG_EDIFACT = "<edifact>\n";
+  public static final String TAG_EDIFACT_END = "</edifact>";
+  public static final String TAG_ELEMENT = "\t\t<element>\n";
+  public static final String TAG_ELEMENT_END = "\t\t</element>\n";
+  public static final String TAG_VALUE = "\t\t\t<value>";
+  public static final String TAG_VALUE_END = "</value>\n";
+
+  public LinkedList<Object> tagIndexes = new LinkedList<Object>();
+
+  // helper functions to sanitize incoming input
+  public String sanitizeText( String txt ) {
+
+    // resolve all RELEASE characters
+    if ( txt.indexOf( "?" ) >= 0 ) {
+      txt = txt.replace( "?+", "+" );
+      txt = txt.replace( "?:", ":" );
+      txt = txt.replace( "?'", "'" );
+      txt = txt.replace( "??", "?" );
+    }
+
+    // enocde XML entities
+    return StringEscapeUtils.escapeXml( txt );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

escapeJava(), escapeHtml3(), escapeHtml4(), escapeXml(), escapeXml10(), and escapeXml11() have been deprecated in org.apache.commons.lang.StringEscapeUtils and org.apache.commons.lang3.StringEscapeUtils and do not cover the entire range of characters that have to be encoded. OWASP Java Encoder library should be used instead.